guidelines: introduce docu, Gl 3.7: physical properties

riedde opened the pull request:
This PR is a proposal by @music-encoding/ig-metadata
The authors of the texts are referenced by the `Co-authored-by` keyword.

_Please note:_ This PR is going to be merged into the branch `ig-metadata`!

Reviews and constructive suggestions are welcome!

ahankinson commented on source/docs/03-metadata.xml:
```diff
@@ -1444,6 +1444,82 @@
                      <p>Watermark</p>
                      <p>Stamp</p>
                      <p>physMedium</p>-->
+                     <p>Encoding the exent and the dimensions of a source:</p>
```
```suggestion
                     <p>Encoding the extent and dimensions of a source:</p>
```

Should this be a header?

ahankinson commented on source/docs/03-metadata.xml:
```diff
@@ -1444,6 +1444,82 @@
+                     <p><specList>
+                        <specDesc key="extent"/>
+                     </specList></p>
+                     <p>The element <gi scheme="MEI">extent</gi> is used to express size in terms other than physical dimensions. This can be done in units such as ‘number of pages’, ‘number of folios’, ‘records’, ‘bytes’, ‘physical components’, etc. For specifying the physical size of the material, for example, height and width, the use of the <gi scheme="MEI">dimensions</gi> element is recommen ...
```
Number of pages is a "physical" dimension (more pages = thicker book.) Perhaps:

```suggestion
                     <p>The element <gi scheme="MEI">extent</gi> is used to express size in units such as ‘number of pages’, ‘number of folios’, ‘records’, ‘bytes’, ‘physical components’, etc. For specifying the physical dimensions of the material&emdash;for example, height and width&emdash;the use of the <gi scheme="MEI">dimensions</gi> element is recommended.</p>
```

ahankinson commented on source/docs/03-metadata.xml:
```diff
@@ -1444,6 +1444,82 @@
+                     <p><specList>
+                        <specDesc key="dimensions"/>
+                     </specList></p>
+                     <p>The <gi scheme="MEI">dimensions</gi> contain information about the physical size of a source or entity in general. Usually the dimensions are represented by numerical data. The elements <gi scheme="MEI">height</gi>, <gi scheme="MEI">width</gi>, <gi scheme="MEI">depth</gi>, and <gi scheme="MEI">dim</gi> are available for circumstances that require the capture of the individ ...
```
As an editorial comment, the use of "...in general" is not needed. It tends to detract from the directness of a statement.

ahankinson replied:
```suggestion
                     <p>The <gi scheme="MEI">dimensions</gi> contain information about the physical size of a source. Usually the dimensions are represented by numerical data. The elements <gi scheme="MEI">height</gi>, <gi scheme="MEI">width</gi>, <gi scheme="MEI">depth</gi>, and <gi scheme="MEI">dim</gi> are available for circumstances that require the capture of individual dimensions.</p>
```

I removed the last sentence because unless you're going to provide an example of how it might be confused, you're only inviting the user to be confused by not knowing how they can be confused by it!

musicEnfanthen replied:
Maybe something like "To indicate the quantity of the described materials, please refer to the extent element" ...?

riedde replied:
Concerning the first deletion: I tried to explain that `<dimensions>` is not limited to a (musical) source, because it can also occur, e.g., at `<watermark>`.

Concerning the second: I see, but I still think it is important to point out the difference between dimensions and extent. 

```suggestion
                     <p>The <gi scheme="MEI">dimensions</gi> contain information about the physical size of a source. Usually the dimensions are represented by numerical data. The elements <gi scheme="MEI">height</gi>, <gi scheme="MEI">width</gi>, <gi scheme="MEI">depth</gi>, and <gi scheme="MEI">dim</gi> are available for circumstances that require the capture of individual dimensions. To indicate the quantity of the described materials, please refer to the extent element.</p>
```

ahankinson commented on source/docs/03-metadata.xml:
```diff
@@ -1444,6 +1444,82 @@
+                     <p><specList>
+                        <specDesc key="watermark"/>
+                     </specList></p>
+                     <p> Before starting to describe and encode watermarks, it is highly recommended to read up on filigranology separately. It is important to distinguish whether the watermark to be described consists of only one sign or is composed of main and countermarks. The completeness of the watermark, its positioning on the manuscript page, and the time period of the paper's production a ...
```
This doesn't really belong in the MEI encoding guidelines. We have to assume that people know what they're doing; otherwise it comes across as patronizing. Focus instead on how you would "properly" encode it, using the correct watermark terminology.

riedde replied:
I think there was an other intention with this paragraph
ping @gucl-mu

riedde replied:
Done by suggestion of @musicEnfanthen

riedde commented on source/docs/03-metadata.xml:
```diff
@@ -1444,6 +1444,82 @@
+                     <p><specList>
+                        <specDesc key="dim"/>
+                     </specList></p>
+                     <p>Another way of encoding is to use the element <gi scheme="MEI">dim</gi>, that can be used to represent any single dimensional specification. The attribute <att>form</att> is required. When appropriate the elements <gi scheme="MEI">height</gi>, <gi scheme="MEI">width</gi>, and <gi scheme="MEI">depth</gi> are preferred to specify the content of <gi scheme="MEI">dim</gi>.</p>
```
```suggestion
                     <p>Another way of encoding is to use the element <gi scheme="MEI">dim</gi>, which can be used to represent any dimensional specification in a generic way. The attribute <att>form</att> is required. With regard to structured metadata, the use of the <gi scheme="MEI">height</gi>, <gi scheme="MEI">width</gi>, and <gi scheme="MEI">depth</gi> elements as the content of <gi scheme="MEI">dim</gi> is advisable.</p>
```

When I read through the guidelines sometimes it is not clear to me which goal is pursued with the guidelines. Is it just the objective explanation of the elements (A can do this and B can do that) or is it also to convey more content (the difference between A and B is...)? Furthermore, I often ask myself whether it is appropriate to show directions (such as "Practice Recommendations")

ahankinson replied:
I think all are fine, but there needs to be a clear indication of whose practice you’re recommending, and why, if you are going to do that. In most cases it is probably clearer to indicate the benefits and compromises, rather than a specific recommendation. Or if something is there for a very specific reason and should not otherwise be used even if it seems, at first sight, to be useful. (MEI is full of these things, where things are allowed inside other things only because they meet a very specific need…)

riedde replied:
Thank you for this insight. This helps me a lot!

musicEnfanthen commented:
@riedde @gucl-mu Is this PR ready for re-review?

bwbohl committed:
Update CONTRIBUTING.md

Co-authored-by: Andrew Hankinson <andrew.hankinson@rism.digital>

bwbohl committed:
Update deploy.yml

* use docker-mei for building schema and guidelines
* copy build assets to schema

bwbohl committed:
Update deploy.yml

* fix typo in ant call

ahankinson committed:
Merge pull request #1118 from music-encoding/pdf-guidelines-test

Implement new build workflow and add PDF Guidelines to build

bwbohl committed:
Fix docker-mei ant command in deploy workflow

ahankinson committed:
Merge pull request #1121 from music-encoding/fix/worflow-build

Fix docker-mei ant command in deploy workflow

bwbohl committed:
fix build warning for local git properties when run in GHAction

* only execute get-local-git-sha and get-local-git-branch when not submitted externally

bwbohl committed:
fix branch argument passed to guidelines xslt

* choose conditionally between github.ref-name and github.branch.local

bwbohl committed:
linting

ahankinson committed:
Fixed: Clarify use of GG clef shape

This is an alternative PR to address the changes requested in #1044.

kepper committed:
Merge pull request #1123 from music-encoding/fix/git-sha-and-ref

Fix: git sha and ref

ahankinson committed:
Moved description to remarks

musicEnfanthen committed:
fix(source): replace umlaut in file name

kepper committed:
spread out constraints to affected elements

fiixes #1129

kepper committed:
remove combining reverse solidus example

this has caused headache in the generation of the Guidelines ever since, and more proper ways to encode exist using the rend element. Thus, dropping this example from the guidelines, replacing it with an alternative.

musicEnfanthen committed:
fix(build): move schemata into correct folder and add canonicalized.xml

musicEnfanthen committed:
feat(gh-actions): upload artifacts on action run

bwbohl committed:
build.xml: add target for building customization specs

bwbohl committed:
add description to build-customization-specs

bwbohl committed:
call build-compiled-odd for specified customization file before building specs

lpugin committed:
Merge pull request #1137 from music-encoding/ftr/build-customization-specs

build.xml: add target for building customization specs

bwbohl committed:
Merge pull request #1134 from musicEnfanthen/fix_deploy_action

fix(build): move schemata into correct folder and add canonicalized.xml

ahankinson committed:
Merge pull request #1136 from musicEnfanthen/add_artifacts_to_deploy_action

feat(gh-actions): upload artifacts on action run

bwbohl committed:
Merge pull request #1105 from musicEnfanthen/patch-neumes-chapter

fix(guidelines): clean up and clarification for neumes chapter

ahankinson committed:
Fix deployment action breaking

Fixes an issue where the directory already exists before moving a new directory to that location.

bwbohl committed:
Merge pull request #1127 from musicEnfanthen/fix_umlaut

fix(source): replace umlaut in file name

jcdevaney committed:
adding extData

Defining <extData> (based on <extMeta>), extending <when> to allow for it to contain <extData>, and updating relevant documentation.

ahankinson committed:
Merge pull request #1116 from rettinghaus/dev/artic

Add stress and unstress to data.ARTICULATION

ahankinson committed:
Merge pull request #1106 from musicEnfanthen/patch2-neumes-chapter

fix(guidelines): fix bibliographic references in neumes chapter

bwbohl committed:
Merge pull request #1124 from music-encoding/fixed-1044-gg-clef

Fixed: Clarify use of GG clef shape

kepper committed:
Update .github/workflows/deploy.yml

Co-authored-by: Stefan Münnich <stefan.muennich@unibas.ch>

musicEnfanthen committed:
Merge pull request #1139 from music-encoding/fix-directory-exists

Fix deployment action breaking

lpugin committed:
Remove modules figtable, frbr and msDesc

lpugin committed:
Remove text module

lpugin committed:
Include only persName from namesdates

lpugin committed:
Modify the element included from meiHead

* Remove title and publisher (not in module)
* Add date

lpugin committed:
Change shared elements from except to include

lpugin committed:
Add midi instrDef

lpugin committed:
Allow pgHead and pgFoot in scoreDef

lpugin committed:
Add back `@label` and `@type` in att.common

kepper committed:
Update source/docs/10-analysisharm.xml

Co-authored-by: Andrew Hankinson <andrew.hankinson@rism.digital>

lpugin committed:
Remove date from header inclusion list (not in module)

ahankinson committed:
Merge pull request #933 from rettinghaus/develop-instrDef

adapt instrDef to staffDef

ahankinson committed:
Fixed: Add glyph.auth and glyph.uri to rend

Addresses result of conversation in #704.

The att.extSym class was broken into two sub-classes so that we could only add rend to one of them, without also pulling in glyph.name and glyph.num.
No other elements should be affected by this change.

lpugin committed:
Change the definition of att.common to include directly att.id

kepper committed:
remove empty attribute classes

musicEnfanthen committed:
Merge pull request #1132 from music-encoding/modify-solidus-example

remove combining reverse solidus example

musicEnfanthen committed:
fix(neumes): fix incorrect attribute classes in neumes customization

bwbohl committed:
Merge pull request #1147 from musicEnfanthen/fix-neumes-customization

fix(neumes): fix incorrect attribute classes in neumes customization

jcdevaney committed:
Update source/docs/12-facsimilesrecordings.xml

Co-authored-by: Benjamin W. Bohl <b.w.bohl@gmail.com>

jcdevaney committed:
Update source/docs/12-facsimilesrecordings.xml

Co-authored-by: Benjamin W. Bohl <b.w.bohl@gmail.com>

ahankinson committed:
Merge pull request #1130 from music-encoding/fix-module-level-schematron

spread out constraints to affected elements

ahankinson committed:
Add an example to the text rendition guidlines

lpugin committed:
Cleanup Verovio examples

pe-ro committed:
Merge pull request #1141 from lpugin/develop-basic-cleanup

Develop basic cleanup

kepper committed:
rename attribute reference in altId remark

ahankinson committed:
Add "fences" as a notehead modifier

This PR changes the notehead modifier list to add the value of `fences` (and removes `dblwhole`). This is as a result of the discussions in #867 and discussions at the MEI Developers Meeting 25.05.2023. 

Refs #865.

ahankinson committed:
Merge pull request #1149 from music-encoding/fix-altId-remark

rename attribute reference in altId remark

ahankinson committed:
Update source/modules/MEI.externalsymbols.xml

Co-authored-by: Stefan Münnich <stefan.muennich@unibas.ch>

bwbohl committed:
Merge pull request #1150 from music-encoding/alt-867-fences

Add "fences" as a notehead modifier

kepper committed:
allow tstamp2.ges and tstamp2.real on metaMark

In #797 and #767 we were too fast in removing `@tstamp2.ges` and `@tstamp2.real` from `<metaMark>`, which was found when evaluating the diff of v4 and v5. Adding them back in, as agreed on the Charlottesville dev meeting.

musicEnfanthen committed:
fix(build): include oxygen-basic-utilites in lib

pe-ro committed:
Word-smithing the description

Co-authored-by: Johanna Devaney <jcdevaney@gmail.com>

pe-ro committed:
Apply suggestions from code review

Co-authored-by: Stefan Münnich <stefan.muennich@unibas.ch>

pe-ro committed:
Merge pull request #1140 from jcdevaney/fix-1131-extData

feat(performance): adding extData

pe-ro committed:
Merge pull request #1151 from music-encoding/fix-metaMark-tstamp2

allow tstamp2.ges and tstamp2.real on metaMark

lpugin committed:
Merge commit 'c7feeb93b60a38d21a0633a17218e5ef544b0217' into develop-coda-segno

# Conflicts:
#	source/modules/MEI.shared.xml

lpugin committed:
Remove att.regularized (back after merge)

lpugin committed:
Remove `other` from the list of possible values for `repeatMark@func`

lpugin committed:
Adjust the descriptions in att.repeatMark.log@func

lpugin committed:
Adjust the description of repeatMark

* Clarify that textual content takes precedence over the symbol implied by the function

lpugin committed:
Update source/modules/MEI.gestural.xml

Co-authored-by: Stefan Münnich <stefan.muennich@unibas.ch>

lpugin committed:
Update source/modules/MEI.gestural.xml

Co-authored-by: Stefan Münnich <stefan.muennich@unibas.ch>

lpugin committed:
Update source/modules/MEI.shared.xml

Co-authored-by: Stefan Münnich <stefan.muennich@unibas.ch>

musicEnfanthen committed:
Merge pull request #1142 from music-encoding/add-glyph-rend

Fixed: Add glyph.auth and glyph.uri to rend

Commit:
fix data.PERCENT

ahankinson committed:
Fixed: instrDef was misspelled

The instrDef element was requested as part of MEI Basic, but it was misspelled so it was not included.

Fixes #1112

ahankinson committed:
Merge pull request #1154 from rettinghaus/fix/percent

fix data.PERCENT

musicEnfanthen committed:
Merge pull request #1155 from music-encoding/fix-1112-instrdef

Fixed: instrDef was misspelled

bwbohl committed:
Merge pull request #1148 from lpugin/develop-fix-examples

Cleanup Verovio examples

ahankinson committed:
Fixed: Use value of "schematron" for constraintSpec

The TEI has changed the value of "schematron" in its RNG.

If we're thinking about moving towards pure ODD this change will mean one less thing that breaks.

Also fixed an issue where the customizations were using the TEI RNG, but the sources were pointing to the (older) MEI RNG. These were normalized to the MEI RNG so that all source and customizations use the same ODD schema.

Fixes #1158

musicEnfanthen committed:
Merge pull request #1022 from lpugin/develop-coda-segno

A expanMark element for encoding coda, segno and similar marks

ahankinson committed:
Fixed: Add copyright notice and update dates

This PR fixes the dates for the copyright notices and also adds the notice to the MEI source files.

It also fixes some formatting issues with the guidelines documents and adds an XML declaration (as noticed by @rettinghaus).

Fixes #1144

kepper committed:
Merge pull request #1161 from music-encoding/fixed-1158-isoschematron

Fixed: Use value of "schematron" for constraintSpec

musicEnfanthen committed:
feat(BUILD_COMMANDLINE): add a list and description of the available ant targets

musicEnfanthen committed:
fix(BUILD_COMMANDLINE): clarify to use absolute path to specific customizations

musicEnfanthen committed:
fix(BUILD_COMMANDLINE): fix path for guidelines html

musicEnfanthen committed:
fix(build.xml): fix typo

musicEnfanthen committed:
fix(guidelines): fix spacing of attributeUsage in by class and by module tabs in HTML guidelines

musicEnfanthen committed:
fix(guidelines): remove duplicated rule for spacing of attributeUsage in PDF guidelines

The duplicated rule reduced the space between the attributeName, attributeUsage and attributeDesc too much.

ahankinson committed:
Update examples with key signature changes

ahankinson committed:
Fixed: Modify key signature attributes

This PR addresses the changes requested in #525. It clarifies the drawing of cancellation accidentals when a new key signature is created,
and makes attribute names more consistent.

Fixes #525

kepper committed:
add validation for customizations

musicEnfanthen committed:
Update BUILD_COMMANDLINE.md

Co-authored-by: Andrew Hankinson <andrew.hankinson@gmail.com>

ahankinson committed:
Update guideline copyright statements

Missed these on the first pass.

ahankinson committed:
Merge pull request #1168 from musicEnfanthen/guidelines-improve-spacing

fix(guidelines): improve spacing of attribute usage

pe-ro committed:
Allow <availability> within <imprint>

Remove <availability> from membership in model.titlePagePart and include it in model.imprintPart. Also make <unpub>a member of model.imprintPart.

rettinghaus committed:
harmonize gestural data types

ahankinson committed:
Merge branch 'develop' into fixed-1144-copyright

musicEnfanthen committed:
fix(build): remove xml declaration from example mei files

musicEnfanthen committed:
fix(source): fix typos

ahankinson committed:
Merge pull request #1179 from musicEnfanthen/fix-typos

fix(source): fix typos

ahankinson committed:
Merge pull request #1178 from musicEnfanthen/fix-java-parsing-errors

fix(build): remove xml declaration from example mei files

ahankinson committed:
Fixed descriptions

ahankinson committed:
Update source/modules/MEI.xml

ahankinson committed:
Merge branch 'develop' into fixed-525-keysig

ahankinson committed:
Fixed: CMN Module was not included in Mensural

No need to remove these classes since they were defined on the CMN
module, and it was not included.

ahankinson committed:
Fixed: Neumes customization

The CMN module was not included, so model classes did not need to be deleted.

Also fixed the name of an attribute class.

ahankinson committed:
Fixed: MEI Basic analytical class

classSpecs are not valid in the "include"; only elementSpec.

ahankinson committed:
Fixed: Elements not in header

The composer, arranger, and lyricist elements are defined in shared, not header.

ahankinson committed:
Fixed: Various misspellings and missing classes

ahankinson committed:
Fixed: classSpec is commented out in the source

these classSpec are commented out, so it was commented out here.

ahankinson committed:
Add Laurent to respStmt and an entry for the dev conference

ahankinson committed:
Fixed: Add note about score redefinition

Also includes an example that should be rendered with Verovio.

ahankinson committed:
Minor doc touchup.

ahankinson committed:
More wordsmithing

ahankinson committed:
Update source/modules/MEI.xml

Co-authored-by: Laurent Pugin <lxpugin@gmail.com>

lpugin committed:
Initial script for generating images using a python script instead of node

lpugin committed:
Update documentation

lpugin committed:
Update utils/examples/generate-examples.py

Co-authored-by: Andrew Hankinson <andrew.hankinson@gmail.com>

lpugin committed:
Update utils/examples/generate-examples.py

Co-authored-by: Andrew Hankinson <andrew.hankinson@gmail.com>

lpugin committed:
Update utils/examples/generate-examples.py

Co-authored-by: Andrew Hankinson <andrew.hankinson@gmail.com>

lpugin committed:
Update utils/examples/generate-examples.py

Co-authored-by: Andrew Hankinson <andrew.hankinson@gmail.com>

lpugin committed:
Update utils/examples/generate-examples.py

Co-authored-by: Andrew Hankinson <andrew.hankinson@gmail.com>

lpugin committed:
Update utils/examples/generate-examples.py

Co-authored-by: Andrew Hankinson <andrew.hankinson@gmail.com>

lpugin committed:
Update utils/examples/generate-examples.py

Co-authored-by: Andrew Hankinson <andrew.hankinson@gmail.com>

lpugin committed:
Update utils/examples/generate-examples.py

Co-authored-by: Andrew Hankinson <andrew.hankinson@gmail.com>

lpugin committed:
Update utils/examples/generate-examples.py

Co-authored-by: Andrew Hankinson <andrew.hankinson@gmail.com>

bwbohl committed:
Merge pull request #1171 from music-encoding/add-customization-validation

add validation for customizations

lpugin committed:
Remove the lxml dependency

* Since having local options is currently not used we can remove the lxml dependency for now
* The script will also fail if and example cannot be loaded or if the resulting SVG is empty

ahankinson committed:
Fixed: Comment out missing class

The model.scorePart.mensuralAndNeumes class is commented out in the spec, so it's commented out here.

lpugin committed:
Apply @ahankinson patch

bwbohl committed:
Merge pull request #1152 from musicEnfanthen/fix-build-on-win

fix(build): include oxygen-basic-utilites in lib

bwbohl committed:
Merge pull request #1165 from musicEnfanthen/update-build-commandline

feat(build): describe ant targets in BUILD_COMMANDLINE

lpugin committed:
Merge commit '1a12eb5e7df4f1da5d3eb29546e7fe18c1c34e6e' into develop-verovio-script

# Conflicts:
#	BUILD_COMMANDLINE.md

lpugin committed:
Update build.xml

Co-authored-by: Stefan Münnich <stefan.muennich@unibas.ch>

lpugin committed:
Update utils/examples/generate-examples.py

Co-authored-by: Stefan Münnich <stefan.muennich@unibas.ch>

pe-ro committed:
Remove "italic" and "oblique" values from data.TEXTRENDITIONLIST

The values "italic" and "oblique" duplicate values available in data.FONTSTYLE.

bwbohl committed:
Merge pull request #1181 from rettinghaus/fix/datatypes

Harmonize gestural data types

lpugin committed:
Update utils/examples/generate-examples.py

Co-authored-by: Andrew Hankinson <andrew.hankinson@gmail.com>

lpugin committed:
Update utils/examples/generate-examples.py

Co-authored-by: Andrew Hankinson <andrew.hankinson@gmail.com>

lpugin committed:
Update utils/examples/generate-examples.py

Co-authored-by: Andrew Hankinson <andrew.hankinson@gmail.com>

lpugin committed:
Update utils/examples/generate-examples.py

Co-authored-by: Andrew Hankinson <andrew.hankinson@gmail.com>

lpugin committed:
Set options only once

martha-thomae committed:
Fix spacing

martha-thomae committed:
Fix examples
Either remove mensuration signs that were not appropriate, add comments to show what the mensuration is in staffDef, fix the implicit-mensuration example (which was changed completely), fix the end tag for verovio in example 'patial-imp-03-remotam.mei'

martha-thomae committed:
Comment out place holder texts

martha-thomae committed:
Add a few words in the introduction section

martha-thomae committed:
Add example for plica

martha-thomae committed:
Add short section for dots

martha-thomae committed:
Update BUILD_OXYGEN.md

Update to the new file path for guidelines

martha-thomae committed:
Update instructions for Java and Ant for Mac users

martha-thomae committed:
Make comments consistent between examples

lpugin committed:
Update source/docs/04-cmn.xml

Co-authored-by: Klaus Rettinghaus <klaus.rettinghaus@gmail.com>

lpugin committed:
Update source/docs/04-cmn.xml

Co-authored-by: Stefan Münnich <stefan.muennich@unibas.ch>

martha-thomae committed:
Update BUILD_COMMANDLINE.md

Co-authored-by: Klaus Rettinghaus <klaus.rettinghaus@enote.com>

martha-thomae committed:
Add brew command to install openjdk

bwbohl committed:
Merge pull request #1172 from music-encoding/develop-imprint-availability

Allow <availability> within <imprint>

bwbohl committed:
Merge pull request #1163 from music-encoding/fixed-1144-copyright

Fixed: Add copyright notice and update dates

musicEnfanthen committed:
Merge pull request #1192 from martha-thomae/contribution_instructions

Update BUILD_OXYGEN.md

bwbohl committed:
build.xml: improve init target

bwbohl committed:
Update build.xml

martha-thomae committed:
Update BUILD_COMMANDLINE.md

Co-authored-by: Benjamin W. Bohl <b.w.bohl@gmail.com>

martha-thomae committed:
Change to more general instructions
Co-authored-by: Benjamin W. Bohl <b.w.bohl@gmail.com>

Commit:
Update source/docs/03-metadata.xml

Commit:
Update source/docs/03-metadata.xml

bwbohl committed:
Update build.xml

Co-authored-by: Klaus Rettinghaus <klaus.rettinghaus@enote.com>

bwbohl committed:
Merge pull request #1183 from music-encoding/fixed-validate-customizations

Fixed: Make customizations valid

bwbohl committed:
Merge pull request #1170 from music-encoding/fixed-525-keysig

Fixed: Key signature changes

lpugin committed:
Convert only `.mei` files

bwbohl committed:
build.xml: introduce property `verbose`

Before this commit the generation of schemata and compiled ODDs gave very verbose feedback. This resulted in a very long and hardly graspable output on the commandline. Especially in the GitHub Action context this made the build process hard to debug and required a lot of scrolling.
This commit introduces the following changes:
* The property `verbose` will analyze whether Ant was called with the `-verbose` or `-v` option and store this as boolean; defaults to false.
* The `verbose` property will be propagated to the  `tei:anttasks` to make the verbosity of the schema and compiled-odd generation controllable.

bwbohl committed:
build.xml: fix location of verbose property condition

bwbohl committed:
build.xml: apply `quiet=true` for `loadresources` globally

* i.e. the GitHub properties to avoid irritating zero-length-warning when run locally

bwbohl committed:
Update BUILD_COMMANDLINE.md

Co-authored-by: Stefan Münnich <stefan.muennich@unibas.ch>

musicEnfanthen committed:
fix(gh-actions): do not run deploy steps on forks

bwbohl committed:
Update BUILD_COMMANDLINE.md

Co-authored-by: Stefan Münnich <stefan.muennich@unibas.ch>

rettinghaus committed:
using Saxon 12.2 from new GitHub releases

Commit:
Update build.xml

Co-authored-by: Benjamin W. Bohl <b.w.bohl@gmail.com>

bwbohl committed:
Merge pull request #1205 from musicEnfanthen/fix-action

fix(gh-actions): do not run deploy steps on forks

lpugin committed:
Adjust MEI- Basic customization to use att.duration.log for rest

lpugin committed:
Update customizations/mei-basic.xml

Co-authored-by: Klaus Rettinghaus <klaus.rettinghaus@enote.com>

bwbohl committed:
Merge pull request #1206 from rettinghaus/dev/saxon

Update to use Saxon release from GitHub

martha-thomae committed:
Update the introduction part regarding 'proportions'

Co-authored-by: Andrew Hankinson <andrew.hankinson@gmail.com>

martha-thomae committed:
Fix word

Co-authored-by: Andrew Hankinson <andrew.hankinson@gmail.com>

martha-thomae committed:
Correct 'dots' section

Co-authored-by: Andrew Hankinson <andrew.hankinson@gmail.com>

martha-thomae committed:
Update introduction text about note duration

Co-authored-by: Andrew Hankinson <andrew.hankinson@gmail.com>

martha-thomae committed:
Fix introductory section about duration

martha-thomae committed:
Add details to dot section
Defining types of dots and pointing to examples

martha-thomae committed:
Update macOS/Linux instructions

Co-authored-by: Stefan Münnich <stefan.muennich@unibas.ch>

martha-thomae committed:
Fix openjdk lines
Co-authored-by: Stefan Münnich <stefan.muennich@unibas.ch>

lpugin committed:
Add `ending` to MEI-Basic

bwbohl committed:
Merge pull request #1208 from lpugin/develop-basic-ending

Add `ending` to MEI-Basic

bwbohl committed:
Merge pull request #1207 from lpugin/develop-basic-rest-duration

Adjust MEI- Basic customization to use att.duration.log for rest

bwbohl committed:
Merge pull request #1188 from music-encoding/develop-text-rendition

Remove "italic" and "oblique" values from data.TEXTRENDITIONLIST

bwbohl committed:
Merge pull request #1196 from bwbohl/ant/init-improve

Improve init target

bwbohl committed:
Merge pull request #1193 from martha-thomae/contrib_instuctions_cli

Update BUILD_COMMANDLINE.md

martha-thomae committed:
Fix word (rendering)

Co-authored-by: Laurent Pugin <lxpugin@gmail.com>

musicEnfanthen commented on source/docs/03-metadata.xml:
```diff
@@ -1444,6 +1444,82 @@
                      <p>Stamp</p>
                      <p>physMedium</p>-->
+                     <p>Encoding the extent and dimensions of a source:</p>
+                     <p><specList>
+                        <specDesc key="extent"/>
+                     </specList></p>
```
Please check indentation and line breaks of all the blocks with

```
<p><specList>
 ...
</specList></p> 
```

Should be:


```
<p>
    <specList>
 ...
    </specList>
</p> 
```

musicEnfanthen commented on source/docs/03-metadata.xml:
```diff
@@ -1444,6 +1444,82 @@
+                     <p><specList>
+                        <specDesc key="dimensions"/>
+                     </specList></p>
+                     <p>The <gi scheme="MEI">dimensions</gi> contain information about the physical size of a source. Usually the dimensions are represented by numerical data. The elements <gi scheme="MEI">height</gi>, <gi scheme="MEI">width</gi>, <gi scheme="MEI">depth</gi>, and <gi scheme="MEI">dim</gi> are available for circumstances that require the capture of individual dimensions. To indica ...
```
```suggestion
                     <p>The <gi scheme="MEI">dimensions</gi> contain information about the physical size of a source. Usually the dimensions are represented by numerical data. The elements <gi scheme="MEI">height</gi>, <gi scheme="MEI">width</gi>, <gi scheme="MEI">depth</gi>, and <gi scheme="MEI">dim</gi> are available for circumstances that require the capture of individual dimensions. To indicate the quantity of the described materials, please refer to the <gi scheme="MEI">extent</gi> element.</p>
```

musicEnfanthen commented on source/docs/03-metadata.xml:
```diff
@@ -1444,6 +1444,82 @@
+                     <p><specList>
+                        <specDesc key="dim"/>
+                     </specList></p>
+                     <p>Another way of encoding is to use the element <gi scheme="MEI">dim</gi>, which can be used to represent any dimensional specification in a generic way. The attribute <att>form</att> is required. With regard to structured metadata, the use of the <gi scheme="MEI">height</gi>, <gi scheme="MEI">width</gi>, and <gi scheme="MEI">depth</gi> elements as the content of <gi scheme= ...
```
```suggestion
                     <p>Another way of encoding dimensional information about a source is to use the element <gi scheme="MEI">dim</gi>, which can be used to represent any dimensional specification in a generic way. The attribute <att>form</att> is required. With regard to structured metadata, the use of the <gi scheme="MEI">height</gi>, <gi scheme="MEI">width</gi>, and <gi scheme="MEI">depth</gi> elements as the content of <gi scheme="MEI">dim</gi> is advisable.</p>
```

musicEnfanthen commented on source/docs/03-metadata.xml:
```diff
@@ -1444,6 +1444,82 @@
                      <p>Watermark</p>
                      <p>Stamp</p>
                      <p>physMedium</p>-->
```
Remove comments here?

riedde replied:
```suggestion
```

musicEnfanthen commented on source/docs/03-metadata.xml:
```diff
@@ -1444,6 +1444,82 @@
+                     <p><specList>
+                        <specDesc key="stamp"/>
+                     </specList></p>
+                     <p>Stamps can appear in many forms in manuscripts and prints, for example as library stamps, library signatures, postmarks, ownership marks, address stamps or legal notices. The description of the stamp therefore depends on individual, project-specific requirements. However, it is helpful to first consider whether it is sufficient to merely name the occurring stamps or if is  ...
```
```suggestion
                     <p>Stamps can appear in many forms in manuscripts and prints, for example as library stamps, library signatures, postmarks, ownership marks, address stamps or legal notices. The description of the stamp therefore depends on individual, project-specific requirements. However, it is helpful to first consider whether it is sufficient to merely name the occurring stamps, or wether it is also desirable to detail their form and textual elements, or even refer to a graphic that shows a facsimile of the stamp.</p>
```

musicEnfanthen commented on source/docs/03-metadata.xml:
```diff
@@ -1444,6 +1444,82 @@
+                           <head/>
+                           <egXML xmlns="http://www.tei-c.org/ns/Examples" rend="code" xml:space="preserve"><xi:include xmlns:xi="http://www.w3.org/2001/XInclude" href="../examples/header/header-sample084.txt" parse="text"/></egXML>
+                        </figure>
+                     </p>
```
`physMedium` missing here? To be added in next release @riedde ?

musicEnfanthen commented on source/docs/03-metadata.xml:
```diff
@@ -1444,6 +1444,82 @@
+                           <egXML xmlns="http://www.tei-c.org/ns/Examples" rend="code" xml:space="preserve"><xi:include xmlns:xi="http://www.w3.org/2001/XInclude" href="../examples/header/header-sample080.txt" parse="text"/></egXML>
+                        </figure>
+                     </p>
+                     <p>For a detailed description or encoding of a watermark, <gi scheme="MEI">fig</gi> can be used, which on the one hand allows reference to an existing image (<gi scheme="MEI">graphic</gi>) of the watermark, and on the other increases the depth of the description. To mark up different components of a watermark, several <gi scheme="MEI">heraldry</gi> elements can be used in a < ...
```
```suggestion
                     <p>For a more detailed description or encoding of a watermark, <gi scheme="MEI">fig</gi> can be used, which on the one hand allows reference to an existing image (<gi scheme="MEI">graphic</gi>) of the watermark, and on the other increases the depth of the description. To mark up different components of a watermark, several <gi scheme="MEI">heraldry</gi> elements can be used in a <gi scheme="MEI">figDesc</gi> and related to each other by means of <gi scheme="MEI">relation</gi>.
```

musicEnfanthen commented on source/docs/03-metadata.xml:
```diff
@@ -1444,6 +1444,82 @@
+                     <p><specList>
+                        <specDesc key="watermark"/>
+                     </specList></p>
+                     <p> Before starting to describe and encode watermarks, it is highly recommended to read up on filigranology separately. It is important to distinguish whether the watermark to be described consists of only one sign or is composed of main and countermarks. The completeness of the watermark, its positioning on the manuscript page, and the time period of the paper's production a ...
```
```suggestion
                     <p> The element <gi scheme="MEI">watermark</gi> can be used to describe watermarks or similar devices of filigranology. It is important to distinguish whether the watermark to be described consists of only one sign or is composed of main and countermarks. The completeness of the watermark, its positioning on the manuscript page, and the time period of the paper's production and use also play a role.</p>
```

musicEnfanthen commented on source/docs/03-metadata.xml:
```diff
@@ -1444,6 +1444,82 @@
+                     <p>
+                        <figure>
+                           <head/>
+                           <egXML xmlns="http://www.tei-c.org/ns/Examples" rend="code" xml:space="preserve"><xi:include xmlns:xi="http://www.w3.org/2001/XInclude" href="../examples/header/header-sample080.txt" parse="text"/></egXML>
```
The new sample080.txt is overwriting the existing sample for the scoreFormat that is used later on (line 1528).

bwbohl replied:
This has to be carefully addressed when resolving conflicts and merging.

musicEnfanthen replied:
@riedde I checked with your ig-metadata branch. There the `scoreFormat` sample file has the number 94.

musicEnfanthen commented on source/examples/header/header-sample080.txt:
The new sample080.txt is overwriting the existing sample for the scoreFormat that is used later on (line 1528).

bwbohl replied:
This has to be carefully addressed when resolving conflicts and merging.

musicEnfanthen commented on source/docs/03-metadata.xml:
```diff
@@ -1444,6 +1444,82 @@
+                           <egXML xmlns="http://www.tei-c.org/ns/Examples" rend="code" xml:space="preserve"><xi:include xmlns:xi="http://www.w3.org/2001/XInclude" href="../examples/header/header-sample081.txt" parse="text"/></egXML>
+                        </figure>
+                     </p>
+                     <p>To ensure that the description of the pictorial signs conforms to international standards and that the individual components of the watermark are correctly represented, the multilingual description catalogue of the <ref target="https://www.memoryofpaper.eu/BernsteinPortal/" type="_blank">Bernstein project</ref> or the <ref target="http://www.paperhistory.org/Standards/" ty ...
```
```suggestion
                     <p>To ensure that the description of the pictorial signs conforms to international standards and that the individual components of the watermark are correctly represented, the multilingual description catalogue of the <ref target="https://www.memoryofpaper.eu/BernsteinPortal/" type="_blank">Bernstein project</ref> or the <ref target="http://www.paperhistory.org/Standards/" type="_blank">IPH standard</ref> should be consulted. To refer to already existing databases with watermarks, see <ptr target="#linkingdata"/>.
```

bwbohl committed:
Revert "Remove "italic" and "oblique" values from data.TEXTRENDITIONLIST"

kepper committed:
add comparison xslt

musicEnfanthen committed:
Merge pull request #1200 from bwbohl/ant/verbosity-control

build.xml: introduce property `verbose`

bwbohl committed:
Merge pull request #1215 from music-encoding/add-comparison-xslt

add comparison xslt

musicEnfanthen committed:
Merge pull request #1214 from music-encoding/revert-1188-develop-text-rendition

Revert "Remove "italic" and "oblique" values from data.TEXTRENDITIONLIST"

rettinghaus committed:
align description with Guidelines

rettinghaus committed:
minor improvement

musicEnfanthen committed:
fix(build): adjust version numbers in BUILD_COMMANDLINE readme

bwbohl committed:
build.xml: check for prince

closes #1201

bwbohl committed:
Merge pull request #1184 from lpugin/develop-verovio-script

Initial script for generating images using a python script instead of…

bwbohl committed:
Update source/docs/03-metadata.xml

Co-authored-by: Stefan Münnich <stefan.muennich@unibas.ch>

Commit:
Update source/docs/03-metadata.xml

Co-authored-by: Stefan Münnich <stefan.muennich@unibas.ch>

Commit:
Update source/docs/03-metadata.xml

Co-authored-by: Stefan Münnich <stefan.muennich@unibas.ch>

Commit:
Update source/docs/03-metadata.xml

Co-authored-by: Stefan Münnich <stefan.muennich@unibas.ch>

Commit:
Update source/docs/03-metadata.xml

Co-authored-by: Stefan Münnich <stefan.muennich@unibas.ch>

Commit:
Update source/docs/03-metadata.xml

Co-authored-by: Stefan Münnich <stefan.muennich@unibas.ch>

bwbohl committed:
deploy.yml: update action/deploy-artifact to v3.1.2

bwbohl committed:
deploy.yml: use glob in PDF asset name

* Using globs in action/upload-artifact is supported since v3.1.2

closes #1125

bwbohl committed:
deploy.yml: whitespace linting

bwbohl committed:
Update .github/workflows/deploy.yml

bwbohl committed:
Update .github/workflows/deploy.yml

rettinghaus committed:
clear up line rendition

Commit:
Update source/docs/02-shared.xml

Co-authored-by: Andrew Hankinson <andrew.hankinson@gmail.com>

Commit:
Update source/docs/02-shared.xml

Co-authored-by: Andrew Hankinson <andrew.hankinson@gmail.com>

ahankinson committed:
Fix caption for curve shape

I noticed that this caption was missing the numeric values. I found the original in the MEI 4 documentation. Not sure if this is an anomaly or a systematic problem.

martha-thomae committed:
Add older sections that were removed
These sections might be empty, but at least they give an idea of other things available in MEI and general information about them, to be filled out later by others. It helps for others to contribute these sections to the guidelines

martha-thomae committed:
Similarly add placeholders for examples for others to fill in

martha-thomae committed:
Fix comment

martha-thomae committed:
Clarification of the provenance of the 'dot' element
The other three following sections are also part of MEI.shared module

martha-thomae committed:
Fix typo

martha-thomae committed:
Merge pull request #1 from martha-thomae/mensuralchap_more

Add back comments lost since v3

Add back commented sections indicating future work. As Anna Plaksin (Mensural IG co-chair) indicated, this information was useful for people to know how accidentals, custos, and coloration work in general or that they are available somewhere in the schema (even if full details are not provided yet). These sections have to be filled out later, but the fact that they exist helps people to have an idea of what is available in the schema and also helps us to allow others to contribute to the guidelines by filling in these sections.

I also used the same style for commenting on places where examples are expected to be provided by the musicologists of the community.

Commit:
Update source/docs/03-metadata.xml

martha-thomae committed:
Add reference to mensuration and proportions sections

musicEnfanthen commented on source/docs/03-metadata.xml:
```diff
@@ -1441,9 +1441,82 @@
                      <p>Extent and Dimension</p>
-                     <p>Watermark</p>
-                     <p>Stamp</p>
-                     <p>physMedium</p>-->
```
@riedde I think you deleted the closing part of the HTML comment tag with your latest commit. Now everything that follows is commented out.

martha-thomae committed:
Update source/docs/05-mensural.xml

Co-authored-by: Stefan Münnich <stefan.muennich@unibas.ch>

musicEnfanthen committed:
fix(guidelines): bring back at-symbol for attributes in PDF guidelines

musicEnfanthen committed:
fix(guidelines): fix style typo in chapter 1

lpugin committed:
Allow `breaksec` in MEI-Basic

musicEnfanthen committed:
Merge pull request #1233 from lpugin/develop-basic-breaksec

customizations: allow `breaksec` in MEI-Basic

bwbohl committed:
Merge pull request #1226 from music-encoding/fix-curve-caption

guidelines: fix caption for curve shape

bwbohl committed:
Merge pull request #1191 from martha-thomae/chap5_fix_v5

guidelines: fix Chapter 5

bwbohl committed:
Merge pull request #1220 from musicEnfanthen/fix-build-cl-readme

docs: adjust version numbers in BUILD_COMMANDLINE readme

musicEnfanthen committed:
Merge pull request #1221 from music-encoding/1201-build-only-run-prince-if-command-found

build: check for prince

bwbohl committed:
Merge pull request #1232 from musicEnfanthen/fix-typo

guidelines: fix style typo in chapter 1

musicEnfanthen committed:
Revert "build: check for prince"

bwbohl committed:
Merge pull request #1236 from music-encoding/revert-1221-1201-build-only-run-prince-if-command-found

Revert "build: check for prince"

bwbohl committed:
Merge pull request #1216 from rettinghaus/fix/oct

schema: align description with guidelines

bwbohl committed:
Merge pull request #1225 from rettinghaus/dev/lform

guidelines: clear up description of line rendition

bwbohl committed:
Merge pull request #1231 from musicEnfanthen/fix-pdf

guidelines: bring back at-sign for attributes in PDF guidelines

musicEnfanthen committed:
Merge pull request #1223 from bwbohl/ftr/dynamic-pdf-name

gh-actions: fix hard-coded PDF name in deploy action

Commit:
Merge branch 'develop' into ig-metadata

# Conflicts:
#	source/docs/03-metadata.xml

Commit:
rename filename and update linking

musicEnfanthen commented:
@riedde Thanks. Can you check https://github.com/music-encoding/music-encoding/pull/1095#discussion_r1232780356 please and have a look at the merge conflicts?

Commit:
fix wellformedness

Commit:
Merge branch 'ig-metadata' into gl-3.7/physicalProperties-merged

# Conflicts:
#	source/docs/03-metadata.xml
#	source/examples/header/header-sample094.txt

riedde commented:
Done!

bwbohl reviewed (commented):
is it really intended to replace header-sample080.txt?

riedde commented:
Do you mean changing the Link from sample080.txt to sample096.txt? Yes. In another Branch sample080.txt already exists and would be overwritten, so I changed it here (link and filename).

bwbohl reviewed (approved):
This is so huge and in future development we probably should have a more frequent synchronisation of the IG branch with the develop branch. I will have to trust you with this, but I don't see a problem anyway because there will be a second PR before this gets merged to develop. Let's look there for the actual resulting changes ;-)
